serverless/functions/media-processor-webhook.protected.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+exports.handler = async function (context, event, callback) {
+  const common = require(Runtime.getAssets()['/common.js'].path);
+  const { getStreamMapItem, createErrorHandler } = common(context, event, callback);
+
+  const client = context.getTwilioClient();
+
+  const { StatusCallbackEvent, ExtensionContext } = event;
+
+  switch (StatusCallbackEvent) {
+    case 'recording-failed':
```

Comment:
I wasn't able to fully test this but asked the API engineer if there is a way to force a recording error.

Reply:
We could maybe test this ourselves by temporarily removing the `.protected` status of this endpoint and sending out own fake webhook event.

Reply:
I tested with this:

```
curl -X POST 'twilio-live-interactive-video-XXXX-XXXX-dev.twil.io/media-processor-webhook' \
  -d 'ExtensionContext={"room": {"name": "RMXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"}}' \
  -d 'ErrorMessage=foo' \
  -d 'StatusCallbackEvent=recording-failed'
```

And also modified the webhook to process one of the events that is always fired...I think a "media processor ended" event to make sure some basics worked.

So I think probably the best we can test on our own.